### ISSUE TITLE
Style/HashConversion auto-correct generates invalid code

### ISSUE THREAD

**Issue by gPrado** [rubocop/rubocop#9683]
The `Style/HashConversion` cop may generate invalid code under some specific scenarios, see example below.

--------

## Expected behavior

Given the following code:

```ruby
Hash[[1,2,3].zip]
# => {1=>nil, 2=>nil, 3=>nil} 
```

RuboCop is expected to not register an offense or to auto-correct it with an equivalent code.

## Actual behavior

RuboCop auto-corrects it as follows:

```ruby
[1,2,3].zip.to_h
# ArgumentError (wrong array length at 0 (expected 2, was 1))
```

## RuboCop version

1.12.1 (using Parser 3.0.1.0, rubocop-ast 1.4.1, running on ruby 2.6.6 x86_64-linux)
